scripts.js

```diff
@@ -0,0 +1,79 @@
+var Acme = (function (Acme) {
+  $(document).ready(function (){
+
+// This function does one thing, and returns a promise
+
+var categories;
+var types;
+var products;
+
+$( document ).ready(function() {
```

Comment:
Pretty sure you don't need two `$(document).ready(function`s within each other.... ;)